contributors.d

```diff
@@ -0,0 +1,187 @@
+    bool showAllContributrs; // will ignore the revRange and show all contributors
+    string cwd; // working directory (should be tools)
+}
+
+/// Used to map multiple authors into one
+GitAuthor[GitAuthor] mappings;
+
+/**
+Search all git commit messages within revRange of all D repositories
+Returns: Array that maps each git `Author: ...` line to a GitAuthor
+*/
+auto findAuthors(string revRange, FindConfig config)
+{
+    Appender!(GitAuthor[]) authors;
+    int commits;
+    foreach (repo; ["dmd", "druntime", "phobos", "dlang.org", "tools", "installer"]
+             .map!(r => buildPath(config.cwd, "..", r)))
+    {
+        if (config.refreshTags)
+        {
+            auto cmd = ["git", "-C", repo, "fetch", "--tags", "https://github.com/dlang/" ~ repo.baseName,
+                               "+refs/heads/*:refs/remotes/upstream/*"];
+            auto p = pipeProcess(cmd, Redirect.stdout);
+            enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+        }
+
+        auto cmd = ["git", "-C", repo, "log"];
+        if (!config.showAllContributrs)
+            cmd ~= revRange;
+        if (config.noMerges)
+            cmd ~= "--no-merges";
+
+        auto p = pipeProcess(cmd, Redirect.stdout);
+        scope(exit) enforce(wait(p.pid) == 0, "Failed to execute '%(%s %)'.".format(cmd));
+
+        static immutable Author = "Author: ";
+        p.stdout
+            .byLineCopy
+            .filter!(line => line.startsWith(Author))
+            .tee!(_ => commits++)
+            .map!((line){
+                line.skipOver(Author);
+                return line.readUser;
+            })
+            .filter!(a => a.name != "The Dlang Bot")
+            // merge similar authors into one
+            .map!(a => mappings.get(a, a))
+            .each!(author => authors ~= author);
+    }
+    if (!config.showAllContributrs)
+        stderr.writefln("Looked at %d commits in %s", commits, revRange);
+    else
+        stderr.writefln("Looked at %d commits", commits);
+    return authors.data;
+}
+
+/// Sorts the authors and filters for duplicates
+auto reduceAuthors(GitAuthors)(GitAuthors authors)
+{
+    import std.uni : sicmp;
+    return authors
+            .sort!((a, b) => sicmp(a.name, b.name) < 0)
+            .release
+            .uniq!((a, b) => a.name == b.name);
+}
+
+/// Original: Name <my@email.com>
+GitAuthor readUser(string s)
+{
+    auto ps = s.splitter("<");
+    GitAuthor author = {
+        name: ps.front.to!string.strip,
+        email: ps.dropOne.front.filter!(a => a != '"').until('>').to!string
+    };
+    return author;
+}
+
+/**
+Reads a mappings file which can be used to map multiple authors into one
+Example:
+foo bar <foo@bar.com>: Foo Bar <foobar@email.com>
+*/
+void readMappings(string gitComitMappingFile)
+{
+    if (!gitComitMappingFile.exists)
+        return;
+
+    File(gitComitMappingFile)
+        .byLineCopy
+        // ignore empty lines and comments, but allow `#ponce`
```

Comment:
Yeah there's a line in the `.commitmap` that starts like this, but thinking about it it makes more sense to use an extra whitespace for it.